text_collector_examples/storcli.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+
+# Script to parse StorCLI's JSON output and expose
+# MegaRAID health as Prometheus metrics.
+#
+# Tested against StorCLI 'Ver 1.14.12 Nov 25, 2014'.
+#
+# StorCLI reference manual:
+# http://docs.avagotech.com/docs/12352476
+#
+# Advanced Software Options (ASO) not exposed as metrics currently.
+#
+# JSON key abbreviations used by StorCLI are documented in the standard command
+# output, i.e.  when you omit the trailing 'J' from the command.
+
+import json
+import subprocess
+
+METRIC_PREFIX = 'megaraid_'
+METRIC_CONTROLLER_LABELS = '{{controller="{}", model="{}"}}'
+
+
+def get_store_cli_json():
+    storcli_cmd = ['/opt/MegaRAID/storcli/storcli64', 'show', 'all', 'J']
```

Comment:
I left it hardcoded as I figured someone using this would likely copy and adapt it (as an example), but agree it's nicer to make it configurable. Would you prefer a commandline flag?